yjit/src/core.rs

```diff
@@ -1623,9 +1623,11 @@ impl Context {
 
     /// Stop using a register for a given stack temp.
     pub fn dealloc_temp_reg(&mut self, stack_idx: u8) {
```

Comment:
What is the difference with this and spill?

Reply:
`spill_temp` is lowered to `Insn::Mov` whereas `dealloc_temp_reg` only mutates `ctx.reg_temps` and generates no code.

Reply:
Exactly. It's mostly used for stack slots that are above the stack top. It's not safe to automatically do this thing at `ctx.stack_pop` because that could still be used, but it never crosses YARV instruction boundaries, so we use this mainly at the YARV instruction boundary.